OpenRA.Mods.Common/Traits/World/ResourceClaimLayer.cs

```diff
@@ -111,6 +102,8 @@ public bool IsClaimedByAnyoneElse(Actor self, CPos cell, out ResourceClaim claim
 				// Same claimer:
 				if (claim.Claimer == self) return false;
 
+				if (claim.Claimer.IsDead || claim.Claimer.Disposed) return false;
```

Comment:
I don't think it matters, when the return is false, the new harvester will MakeClaim and overwrite the dead one.